src-cpp/rdkafkacpp.h

```diff
@@ -146,6 +146,40 @@ std::string get_debug_contexts();
 RD_EXPORT
 int          wait_destroyed(int timeout_ms);
 
+/**
+ * @brief Allocate memory using the same allocator librdkafka uses.
+ *
+ * This is typically an abstraction for the malloc(3) call and makes sure
+ * the application can use the same memory allocator as librdkafka for
+ * allocating pointers that are used by librdkafka.
+ *
+ * \p rk can be set to return memory allocated by a specific \c rk instance
+ * otherwise pass NULL for \p rk.
+ *
+ * @remark Memory allocated by mem_malloc() must be freed using
+ *         mem_free()
+ */
+RD_EXPORT
+void *mem_malloc(struct rd_kafka_s *rk, size_t size);
```

Comment:
I think it should take an optional Handle pointer, not an rk, no need to expose the C type here.